models/vrp.rb

```diff
@@ -452,8 +452,12 @@ def self.deduce_unavailable_days(hash, element, start_index, end_index, type)
 
     def self.detect_date_indices_inconsistency(hash)
       missions_and_vehicles = hash[:services].to_a + hash[:shipments].to_a + hash[:vehicles].to_a
-      has_date = missions_and_vehicles.any?{ |m| m[:unavailable_date_ranges] || m[:unavailable_work_date] }
-      has_index = missions_and_vehicles.any?{ |m| m[:unavailable_index_ranges] || m[:unavailable_work_day_indices] }
+      has_date = missions_and_vehicles.any?{ |m|
+        !(m[:unavailable_date_ranges] || m[:unavailable_work_date]).to_a.empty?
+      }
+      has_index = missions_and_vehicles.any?{ |m|
+        !(m[:unavailable_index_ranges] || m[:unavailable_work_day_indices]).to_a.empty?
+      }
```

Comment:
simpler is better
`&.any?` instead of `!.to_a.empty?`
```suggestion
      has_date = missions_and_vehicles.any?{ |m|
        (m[:unavailable_date_ranges] || m[:unavailable_work_date])&.any?
      }
      has_index = missions_and_vehicles.any?{ |m|
        (m[:unavailable_index_ranges] || m[:unavailable_work_day_indices])&.any?
      }
```